### PR TITLE
Add plot recipe for `pks`

### PR DESCRIPTION
**Pull request by MrHenning**
Hi,
I don't know if this is desirable, but I added a plot-recipe for peaks, to be used as
```julia
using Peaks, Plots
t = 0:1/100:1
y = 2*sin.(5*t)+3*sin.(10*t)+2*sin.(30*t)
pks, vals = findmaxima(y)
pks, proms = peakproms!(pks, y; minprom=1)
plt = peaksplot(t, y, peaks=pks)
# or `peaksplot(t, y, peaks=pks, maxima=false)` to plot minima
```
peaks_prom_width

Only `RecipesBase` is added as a (very lightweight) dependency, `Plots.jl` must be installed by the user, or will be installed when running the unit tests.

Heavily inspired by https://github.com/JuliaNLSolvers/LsqFit.jl/pull/180

PS: Thanks for this great package, it has served me extensively in the past :)

### CODE REVIEW

**Comment by halleysfifthinc** [comment]
Thanks for the PR! I rebased on master and made some tweaks to the interface and presentation.

The tests will need to be updated but I think that is all that is needed to merge. I plan on updating the tests tomorrow unless you get to them first! :grin:

**Comment by MrHenning** [comment]
Nice! :slightly_smiling_face: 
I'll try to update the test later today!

**Comment by MrHenning** [comment]
@halleysfifthinc I updated the test, but the horizontal prominence lines are broken in the plot.
So I removed the `drop_irrelevant_side` method, since it works better without it ^^
I updated the `README.md` and `plots/` as well.

If you don't agree with the changes, feel free to revert them, or you can tell me what to do :slightly_smiling_face:

**Review by halleysfifthinc — CHANGES_REQUESTED** [review]
Any ideas how we could avoid unnecessarily duplicating series when making the extrema figure?

**Comment by halleysfifthinc** [comment]
Another reason why I prefer dropping the unneeded side from the prominence lines:

image

I'm not a fan of the left side of the width line just floating, but I don't know what an equivalent solution would be.

**Comment by MrHenning** [comment]
ah ok I understand 😅 
I'll revert the change I made

File: src/plot.jl [hunk 1]
```diff
@@ -12,22 +12,14 @@ function interp(x::AbstractVector{<:Real}, idx::AbstractVector{<:Real})
...
 @recipe function f(::Type{Val{:peaksplot}}, x, y, z; peaks::AbstractVector{Int}, prominences=false, widths=false)
-    if ismaxima(first(peaks), y; strict=false)
+    if all(ismaxima.(peaks, Ref(y); strict=false))
         maxima = true
-    elseif isminima(first(peaks), y; strict=false)
+    elseif all(isminima.(peaks, Ref(y); strict=false))
         maxima = false
     else
-        throw(error("The first peak in `peaks` is not a local extrema"))
+        throw(error("The peaks in `peaks` must *either* local maxima *or* minima."))
```
[the remark this answers is not in the document]
Comment (MrHenning) [reply src/plot.jl]: reverted

File: src/plot.jl [hunk 2]
```diff
@@ -57,13 +49,8 @@ end
         _, _, lower, upper = peakwidths(peaks, y, proms;
             strict=false, relheight=prevfloat(1.0))
 
-        lower_x = interp(x, drop_irrelevant_side.(lower, peaks, Ref(y), maxima))
-        upper_x = interp(x, drop_irrelevant_side.(upper, peaks, Ref(y), maxima))
```
[the remark this answers is not in the document]
Comment (MrHenning) [reply src/plot.jl]: reverted

**Comment by halleysfifthinc** [comment]
One last question: I'm not familiar with the Plots ecosystem (I use PlotlyJS directly). Is a subject-verb phrasing (e.g. `peaksplot`) a common naming convention for recipes? I would otherwise prefer a more imperative, verb-subject phrasing (i.e. `plotpeaks`).

**Comment by MrHenning** [comment]
I would say it’s the more common order, see[StatsPlots.jl](https://github.com/JuliaPlots/StatsPlots.jl)  or [GraphRecipes.jl](https://github.com/JuliaPlots/GraphRecipes.jl) 

But I have no strong opinion on this. `plotpeaks` indeed sounds more intuitive if I think about it.

**Comment by halleysfifthinc** [comment]
Hmm I think a key difference compared to plots in StatsPlots or GraphRecipes is that those are well-known named kinds of plots with proper names, whereas in this case I don't think that a "peaks" plot is a distinct/common named plot format. I'll switch it to `plotpeaks` and merge.

FYI There are another couple of updates I have planned before tagging v0.5.0, so I won't be tagging/releasing immediately.

**Comment by MrHenning** [comment]
Ok, no hurry from my side!

Thanks for the review!!! 😀

**Pull request merged**